### PR TITLE
first try on upgrading to cake 4

### PR DESCRIPTION
**Pull request by gringlas**
Hi everybody, 
I have created an upgrade of this plugin for cake 4.0.* . The phpunit testsuite went through without any errors and I've tested the version on one of my projects. 
This is my first pull request ever, so please bear that in mind, in case I have done anything wrong. Would be great to update this fine plugin though to use it in cake4. 

Thank you and greetings from Germany,
 Sebastian Köller

### CODE REVIEW

**Comment by nojimage** [comment]
Thank you for your great work.

I hasn't been fully reviewed it yet, but it seems that some unnecessary files have been committed.

- .idaa/
- .phpunit.result.cache
- composer.lock
- main

Can you exclude these files from the PR?

**Comment by gringlas** [comment]
Thats good to hear :) 
I've removed alle unnecessary files, but I think the composer.lock needs to stay in version control, like stated here:
https://stackoverflow.com/questions/12896780/should-composer-lock-be-committed-to-version-control
Or am I wrong in that point?

**Comment by nojimage** [comment]
[https://getcomposer.org/doc/02-libraries.md#lock-file](https://getcomposer.org/doc/02-libraries.md#lock-file)
> For your library you may commit the composer.lock file if you want to. This can help your team to always test against the same dependency versions. However, this lock file will not have any effect on other projects that depend on it. It only has an effect on the main project.

It is better not to include composer.lock because the library needs to work in various environments.
It also gets in the way when testing with different PHP/CakePHP versions.

(If your application runs in a single environment, I recommend including composer.lock.

**Comment by gringlas** [comment]
Ok, thanks for the explaination. I've removed composer.lock and added also added it to .gitignore.

File: composer.json [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
     "license": ["MIT"],
     "require": {
-        "php": ">=5.6",
+        "php": ">=7.2",
         "ext-pdo": "*",
         "ext-json": "*",
-        "cakephp/cakephp": "^3.4",
-        "psr/log": "^1.0"
+        "psr/log": "^1.0",
+        "cakephp/cakephp": "4.0.*"
```
Comment (nojimage) [inline composer.json:11]: Please change to like bellow, include CakePHP 4.1.

```suggestion
        "cakephp/cakephp": "^4.0"
```

**Review by nojimage — CHANGES_REQUESTED** [review]
Please change the following points if possible. thanks.

File: src/Model/Behavior/LoggerBehavior.php [hunk 2]
```diff
@@ -485,7 +485,7 @@ private function saveLogs($logs)
         $logTable = $this->getLogTable();
         /* @var ActivityLogsTable $logTable */
         foreach ($logs as $log) {
-            $logTable->save($log, ['atomic' => false]);
+            $logTable->save($log);
```
[the remark this answers is not in the document]
Comment (gringlas) [reply src/Model/Behavior/LoggerBehavior.php:488]: Nope. Can't remember why I did that, but I reverted that change and tests are passing.

File: tests/Fixture/ActivityLogsFixture.php [hunk 3]
```diff
@@ -13,8 +13,8 @@
 class ActivityLogsFixture extends TestFixture
 {
     public $fields = [
-        'id' => ['type' => 'biginteger', 'length' => 20, 'unsigned' => true, 'null' => false, 'default' => null, 'comment' => '', 'autoIncrement' => true, 'precision' => null],
```
[the remark this answers is not in the document]
Comment (gringlas) [reply tests/Fixture/ActivityLogsFixture.php:16]: I'm not exactly sure, but are you refering to the change of type biginteger to integer? I've reverted that change and tests are passing.

**Comment by gringlas** [comment]
I have commited a version with your changes and also removed syntaxCheck atrribute in phpunity.xml.dist as it was removed long ago (https://stackoverflow.com/questions/44328114/phpunit-what-does-syntaxcheck-configuration-parameter-stands-for-exactly)

**Comment by nojimage** [comment]
Thank you for fixes. I'll merge this.

**Pull request merged**